OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ssh/signature/SshSignature.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2017 Christian Hagau <ach@hagau.se>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.sufficientlysecure.keychain.ssh.signature;
+
+import com.android.annotations.Nullable;
+import org.bouncycastle.asn1.ASN1Integer;
+import org.bouncycastle.asn1.ASN1Primitive;
+import org.bouncycastle.asn1.ASN1Sequence;
+import org.bouncycastle.bcpg.PublicKeyAlgorithmTags;
+import org.bouncycastle.util.BigIntegers;
+import org.sufficientlysecure.keychain.pgp.UncachedPublicKey;
+import org.sufficientlysecure.keychain.ssh.key.SshEncodedData;
+import org.sufficientlysecure.keychain.ssh.utils.SshUtils;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.security.NoSuchAlgorithmException;
+
+public class SshSignature {
+
+    private SshEncodedData mSignatureBlob;
+
+    private byte[] mRawSignature;
+
+    private int mAlgorithm;
+
+    private UncachedPublicKey mPublicKey;
+
+    public SshSignature(byte[] rawSignature, int algorithm, @Nullable UncachedPublicKey publicKey) {
+        mSignatureBlob = new SshEncodedData();
+        mRawSignature = rawSignature;
+        mAlgorithm = algorithm;
+        mPublicKey = publicKey;
+    }
+
+    private String getSignatureType() throws NoSuchAlgorithmException {
+        switch (mAlgorithm) {
+            case PublicKeyAlgorithmTags.RSA_SIGN:
+            case PublicKeyAlgorithmTags.RSA_GENERAL:
+                return "ssh-rsa";
+            case PublicKeyAlgorithmTags.ECDSA:
+                return "ecdsa-sha2-"+ SshUtils.getCurveName(mPublicKey);
```

Comment:
Why pass the whole UncachedPublicKey here, *and* the algorithm? Either just pass the public key, or pass the algorithm + curve oid. I'd say the latter is nicer